### PR TITLE
[expo] improve support for flutter native ios dependencies

### LINKED ISSUE #16: native deps
this is great and got me pretty far on a poc I am working on. what about native dependencies the flutter app uses? usually those get auto linked on build, how would this work in an expo project? Any ideas or breadcrumbs?

### PR DESCRIPTION
**Pull request by p-mazhnik**
Fixes https://github.com/p-mazhnik/flutter-embedding/issues/16

### CODE REVIEW

**Comment by joernroeder** [comment]
tried to run this quickly by checking out the `expo-ios-deps` branch but got the following error.

> CommandError: Failed to build iOS project. "xcodebuild" exited with error code 65.
To view more error logs, try building the app with Xcode directly, by opening ./expo-flutter/ios/expoflutter.xcworkspace.
>
> Command line invocation:
    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace ./expo-flutter/ios/expoflutter.xcworkspace -configuration Debug -scheme ExpoFlutter -destination id=665B6112-96DB-4850-A456-D5F05AD521D2
>
> User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES
>
>
> xcodebuild[24047:33845983] Writing error result bundle to /var/folders/1v/ql40bh0n42df7pyrl01d32100000gp/T/ResultBundle_2024-23-09_19-12-0010.xcresult
xcodebuild: error: The workspace named "expoflutter" does not contain a scheme named "ExpoFlutter". The "-list" option can be used to find the names of the schemes in the workspace.

I also noticed that you're removing the `--clean` flag, https://github.com/p-mazhnik/flutter-embedding/pull/17/files#diff-6da0e730d49be64cf3afe029cbe017c8c50fd8cfee58fa2b8ff4bccc2bdfc7d2L113.

does that mean we'll end up with an iOS and android folder again (so far I'm able to clean up on prebuild)?

File: expo-flutter/ios/Podfile [hunk 1]
```diff
@@ -39,6 +39,16 @@ end
 
 target 'ExpoFlutter' do
   use_expo_modules!
+# @generated begin expo-flutter-view - expo prebuild (DO NOT MODIFY) sync-a6fad925eb19db4b427cb412b9bf06f47580494d
+  pod 'Flutter', :podspec => '../modules/expo-flutter-view/ios/Flutter/Flutter.podspec'
+  pod 'FlutterPluginRegistrant', path: '/Users/mazhnik/Codes/p-mazhnik/flutter-embedding/expo-flutter/modules/expo-flutter-view/ios/Flutter/plugins/FlutterPluginRegistrant'
```
Comment (joernroeder) [inline expo-flutter/ios/Podfile]: full path leaked

Comment (p-mazhnik) [reply expo-flutter/ios/Podfile]: Yeah I fixed it in the latest commits, squash and merge will prevent this from going to the main branch. 

> I also noticed that you're removing the --clean flag

I just don't think the `clean` flag is needed for each change. It takes a long time for me to reinstall all pod dependencies when using `clean`, and it is unnecessary to use it every time (although not prohibited)

> xcodebuild: error: The workspace named "expoflutter" does not contain a scheme named "ExpoFlutter". 

hm, I am looking at the PR changes and don't see anything related. I can quickly setup build in GH Actions to check

Comment (p-mazhnik) [reply expo-flutter/ios/Podfile]: Working with scheme written in lower case: https://github.com/p-mazhnik/flutter-embedding/actions/runs/11011828146/job/30576763675
`--scheme expoflutter`

Comment (p-mazhnik) [reply expo-flutter/ios/Podfile]: I have no idea why this is changed by the PR, and locally I still have scheme with upper case letters

Comment (joernroeder) [reply expo-flutter/ios/Podfile]: I think I lost it when I ran prebuild with --clean

**Pull request merged**